src/main/scala/iog/psg/cardano/TxMetadataOut.scala

```diff
@@ -22,19 +22,19 @@ final case class TxMetadataOut(json: Json) {
 
       def extractStringField(cursor: ACursor): DecodingEither[MetadataValueStr] =
```

Comment:
Lines 10 to 16, these should be defined in a companion object and not in every instance of the class.

Reply:
Lines 10 to 16 Maciej, including the type definition.... it does not need to be declared on every instanciation.

Reply:
missed type, I just pushed a commit I see that KeyVal type can be moved there too then, so moved it also